### PR TITLE
UptimeCheckConfig custom delete for 400s

### PR DESCRIPTION
**Pull request by jdurelli**
Add a custom delete method for UptimeCheckConfigs so 400 errors can warn users to delete their Alert Policies first.

Fixes: https://github.com/hashicorp/terraform-provider-google/issues/10176

```release-note:enhancement
monitoring: Added error message for `google_monitoring_uptime_check_config` deletes on 400 response to delete Alert Policies first.
```

### CODE REVIEW

File: mmv1/products/monitoring/UptimeCheckConfig.yaml [hunk 1]
```diff
@@ -72,6 +72,7 @@ custom_code: !ruby/object:Provider::Terraform::CustomCode
   constants: templates/terraform/constants/monitoring_uptime_check_config.go.erb
   custom_import: templates/terraform/custom_import/self_link_as_name.erb
   post_create: templates/terraform/post_create/set_computed_name.erb
+  custom_delete: templates/terraform/custom_delete/monitoring_uptime_check_config.go.erb
```
Comment (ScottSuarez) [inline mmv1/products/monitoring/UptimeCheckConfig.yaml:75]: Adding a custom delete also seems to remove the sweeper

https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-9429-old..auto-pr-9429

Can you maintain the sweeper going forward with a custom sweeper? (likely just copy and paste old code)

File: mmv1/templates/terraform/custom_delete/monitoring_uptime_check_config.go.erb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+<%#
+    The generated DELETE method returns a generic error for 400. 
+    Need to include a help message about deleting associated Alert Policies.
+-%>
+
+if err != nil {
+    if transport_tpg.IsGoogleApiErrorWithCode(err, 400) {
+        err := fmt.Errorf("%w Make sure all associated Alert Policies are deleted first.", err)
```
Comment (ScottSuarez) [inline mmv1/templates/terraform/custom_delete/monitoring_uptime_check_config.go.erb:48]: My worry here is that we are checking a 400 and then saying `Make sure all associated Alert Policies are deleted first`, as though that is the cause of the 400. How do we know that this is the case? If we are not sure maybe we can make the statement less authoritative?

```suggestion
        err := fmt.Errorf("%w - please ensure all Alert Policies are deleted.", err)
```

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/templates/terraform/custom_delete/monitoring_uptime_check_config.go.erb (hunk 2 above)
Comment (jdurelli) [reply mmv1/templates/terraform/custom_delete/monitoring_uptime_check_config.go.erb]: Done

File: mmv1/products/monitoring/UptimeCheckConfig.yaml (hunk 1 above)
Comment (jdurelli) [reply mmv1/products/monitoring/UptimeCheckConfig.yaml:75]: Tried adding it - will see if I did it right

Comment (jdurelli) [reply mmv1/products/monitoring/UptimeCheckConfig.yaml:75]: @ScottSuarez  any idea where the copies of the sweeper file should live? https://googlecloudplatform.github.io/magic-modules/get-started/how-magic-modules-works/ did not seem to point me to the right place

Comment (ScottSuarez) [reply mmv1/products/monitoring/UptimeCheckConfig.yaml:75]: Try adding the file to `mmv1/third_party/terraform/services/monitoring`

Comment (jdurelli) [reply mmv1/products/monitoring/UptimeCheckConfig.yaml:75]: That worked with a little bit of tweaking

**Comment by ScottSuarez** [comment]
Looks like the sweeper file isn't getting copied now

**Comment by ScottSuarez** [comment]
waiting to review durring iteration. Are you able to build locally successfully ?

**Comment by jdurelli** [comment]
> Looks like the sweeper file isn't getting copied now

It is, it's just identical to the existing sweeper so it doesn't appear in the diff

**Comment by jdurelli** [comment]
> waiting to review durring iteration. Are you able to build locally successfully ?

I had no issues building locally.

**Comment by ScottSuarez** [comment]
> > Looks like the sweeper file isn't getting copied now
> 
> It is, it's just identical to the existing sweeper so it doesn't appear in the diff

🤦🏻  makes sense!

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**